### DISCUSSION TITLE
Clearing the status of completed check_runs from Check_suite after Re-Run request on a Check_Run

### DISCUSSION THREAD

**Discussion by cdami** [community/community#38288] [Apps, API and Webhooks] [answered]
Hello,

After reading the docs and some experimentation I am having troubles to find a solution about how to implement a solution for our scenario.

We have a private Github App that currently does receive the check_suite requested events and as a response is creating a number of check_runs under the check_suite.

After the check_runs are marked as completed, we would like the repo users to potentially request a Re-Run of a particular check  that failed. In doing so, other checks in the check_suite shall be reseted as well, not only the rerequested one.

Problem description:
1. when a user clicks **Re-Run** on a check_run we do receive a check_run rerequested event but the check_run is not reseted. Also if we invoke via the API this request we get a 201 but not the desired result (i.e. When a check run is rerequested, its status is reset to queued and the conclusion is cleared. --- [source](https://docs.github.com/en/rest/checks/runs#rerequest-a-check-run)  )
this still shows as completed and the status does not change on Github Web UI
the status still show as completed if getting the check_run via API

2. if we solve 1, we would still need to find a way to clearing the status of a number of other check_runs that would be an implied dependency.

3. For check_runs that have output images I couldn't find a way to reset this (i.e. starting again with no images)

Thank you,
Calin

**Comment by ossanna16** [comment]
Hi there @cdami and welcome to our community! Thank you for asking a great question 🙂
While you are waiting for a reply to your question, here are a few things you can do:

To get started, introduce yourself in our [official introduction thread](https://github.com/orgs/community/discussions/23204)
To connect with other new users, check out this Discussion to [Meet Your GitHub Community Team](https://github.com/orgs/community/discussions/28776), further your GitHub knowledge by working through our [GitHub Skill 🆙 exercises](https://github.com/orgs/community/discussions/32121), and participate in our [weekly Fun Friday chats](https://github.com/orgs/community/discussions/30912)

**Comment by cdami** [comment, marked as answer]
Replying to myself, maybe others are looking to solve the same problem.

1. When a check run is rerequested, its status is reset to queued and the conclusion is cleared --> NOT ; I was able to reset it by recreating a check_run with exactly the same name
2. works for the other check_runs that you need to reset status for with the same solution mentioned at 1.
3. solved in the same way, no matter the previous status of output.images or how you create the check_run, i.e. with output.images = [] or without including it in the payload

I suppose the Github documentation shall be updated to reflect how things work in reality?

Calin
